### PR TITLE
Add LED icon class

### PR DESCRIPTION
**Pull request by dc2917**
This PR adds a new class for handling LED indicators on the GUI. The main advantage is that the image data are loaded only once, rather than for each separate GUI component that requires them. I have also provided some additional functionality, allowing the LED icons to flash on for a specified duration, once some process triggers them (e.g. polling the server).

### CODE REVIEW

File: finesse/gui/led_icons.py
Comment (alexdewar) [inline finesse/gui/led_icons.py:17]: All these variables are public so they'll be exported from the module. I'd make them private (by putting an underscore in front).

File: finesse/gui/led_icons.py
Comment (alexdewar) [inline finesse/gui/led_icons.py:23]: I'd make `status` a `bool` and rename it to something like `is_on` (throughout).

File: finesse/gui/led_icons.py
Comment (alexdewar) [inline finesse/gui/led_icons.py:28]: Rather than defining these as empty images, I'd have `on_img` and `off_img` as arguments to this function, then your child classes can just set these in their calls to `super().__init__`.

File: finesse/gui/led_icons.py
Comment (alexdewar) [inline finesse/gui/led_icons.py:67]: These lines could be in the parent class's `__init__()` instead, if you make the change I suggested above.

File: finesse/gui/led_icons.py
Comment (alexdewar) [inline finesse/gui/led_icons.py:55]: I think `status` should default to off (and in the child classes).

**Review by alexdewar — CHANGES_REQUESTED** [review]
I've made some small suggestions but this looks good :-)

File: finesse/gui/led_icons.py
Comment (alexdewar) [inline finesse/gui/led_icons.py:48]:
```suggestion
        self._is_on = False
```

File: finesse/gui/led_icons.py
Comment (alexdewar) [reply finesse/gui/led_icons.py:51]: What about this? ^^

**Review by alexdewar — CHANGES_REQUESTED** [review]
Found a couple of really minor things.

Something else I probably should have asked earlier is, where did you get these icons from? What licence (if any) are they released under?

File: finesse/gui/led_icons.py
Comment (dc2917) [reply finesse/gui/led_icons.py:48]: > Found a couple of really minor things.

Thanks! I should have been more careful with converting int to bool!

> Something else I probably should have asked earlier is, where did you get these icons from? What licence (if any) are they released under?

Good question - I made them on Inkscape haha so fully free and open source!

**Review by alexdewar — APPROVED** [review]
LGTM.

Although in the commit message for 875e3ee you say you're setting the timeout to 100 ms when you've set it to 250 ms. It doesn't particularly matter though.

**Comment by dc2917** [comment]
> LGTM.
> 
> Although in the commit message for [875e3ee](https://github.com/ImperialCollegeLondon/FINESSE/commit/875e3ee4f241aa9e17efa8d8c412a4f5adb65384) you say you're setting the timeout to 100 ms when you've set it to 250 ms. It doesn't particularly matter though.

Wow, I'm really not with it today 😅

File: finesse/gui/led_icons.py
Comment (dalonsoa) [inline finesse/gui/led_icons.py:61]: I think this and the alarm version should not be subclasses of LEDIcon, but factory methods of LEDicon that create the object with slightly different inputs.

So these two init methods should be converted into two class methods of LEDicon, called constructors. Then, in the tenperature controler, yo use those as:

```python
normalLED = LEDIcon.create_poll_icon()
alarmLED = LEDIcon.create_alarm_icon()
```

In summary, just one class- you just create it in a slightly different way.

**Review by dalonsoa — CHANGES_REQUESTED** [review]
Great improvement of modularity. Just a suggestion to ise constructor.

File: finesse/gui/led_icons.py
Comment (dc2917) [reply finesse/gui/led_icons.py]: Thanks! I knew there must be a neater way haha. I think what I've now done achieves this.

File: finesse/gui/led_icons.py
Comment (dalonsoa) [inline finesse/gui/led_icons.py:49]: The usual approach when dealing with factory methods like these is:
```suggestion
    @clasmethod
    def create_poll_icon(cls) -> LEDIcon:
        """Creates the LED icon for polling the server."""
        return cls(on_img=_poll_on_img, off_img=_poll_off_img)

    @clasmethod
    def create_alarm_icon(cls) -> LEDIcon:
        """Creates the LED icon to indicate alarm status."""
        return cls(on_img=_alarm_on_img, off_img=_alarm_off_img)
```

For this to work, you will need to import at the top of the file:

```python
from __futures__ import annotations
```

**Review by dalonsoa — APPROVED** [review]
Just a small correction, but looks good otherwise!

**Comment by alexdewar** [comment]
Looks like this is being affected by #95. Should be fixed once #86 is merged.

**Comment by alexdewar** [comment]
Would you like me to merge this when it's fixed btw?

**Comment by dc2917** [comment]
> Looks like this is being affected by #95. Should be fixed once #86 is merged.

Yep, I was glad to see it wasn't me breaking something 😅

**Comment by dc2917** [comment]
> Would you like me to merge this when it's fixed btw?

Yes please, thanks!

**Pull request merged**
